h/accounts/__init__.py

```diff
@@ -34,8 +35,25 @@ def make_staff(username):
         raise NoSuchUserError
 
 
+def authenticated_user(request):
+    """Return the authenticated user or None.
+
+    :rtype: h.accounts.models.User or None
+
+    """
+    if not request.authenticated_userid:
+        return None
+
+    # pylint: disable=unpacking-non-sequence
+    username, _ = util.split_user(request.authenticated_userid)
```

Comment:
`request.authenticated_userid` is set by the authentication policy, which is the pyramid default, `pyramid.authentication.SessionAuthenticationPolicy`. You could provide a custom authentication policy which ensured the userid was in the right domain if you want. That might not be a bad idea. Regardless, any code which takes a string userid and turns it into a user object should ensure that it's not just throwing away a domain which doesn't match.

I promise I'm not this retentive about everything. But for security components... better safe than sorry.